commit b4e0fe4c69a5f2ab55a067261c4fa1be93b96d8a (refs/changes/75/2775/4)
Author: Tim Kordas <tim.kordas@joyent.com>
Date:   2017-10-16T14:51:34-07:00 (2 years ago)
    
    joyent/node-cueball#132 Expose counters and simple metrics from CueBallConnectionPool

diff --git a/lib/pool.js b/lib/pool.js
index 84cab63..f827e8b 100644
--- a/lib/pool.js
+++ b/lib/pool.js
@@ -729,6 +729,23 @@ CueBallConnectionPool.prototype.printConnections = function () {
 	console.log('dead:', self.p_dead);
 };
 
+CueBallConnectionPool.prototype.getStats = function () {
+	// Snapshot current counter values.
+	var counters = {};
+	Object.keys(this.p_counters).forEach(function (k) {
+		counters[k] = this.p_counters[k];
+	});
+
+	var stats = {};
+	stats['counters'] = counters;
+	stats['totalConnections'] = Object.keys(this.p_connections).length;
+	stats['idleConnections'] = this.p_idleq.length;
+	stats['pendingConnections'] = this.p_initq.length;
+	stats['waiterCount'] = this.p_waiters.length;
+
+	return (stats);
+};
+
 CueBallConnectionPool.prototype.claim = function (options, cb) {
 	var self = this;
 	var done = false;
